### PR TITLE
Don't show prefix and postfix if there is no placeholder items.

### PR DESCRIPTION
**Pull request by AndyDune**
For show h1 in layout with set in view file.
```
<?= $this->placeholder('h1')->setPrefix('<h1>')->setPostfix('</h1>'); ?>
```

### CODE REVIEW

File: src/Helper/Placeholder/Container/AbstractContainer.php [hunk 1]
```diff
@@ -118,6 +118,10 @@ public function toString($indent = null)
             : $this->getIndent();
 
         $items  = $this->getArrayCopy();
+        // If we don't have items - do not show prefix and postfix
```
Comment (froschdesign) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: If no item is present, then this check should the first step in this method. You can use `$this->count()`.

Comment (froschdesign) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: Please provide also an unit test for this new behaviour. Thanks!

File: src/Helper/Placeholder/Container/AbstractContainer.php [hunk 2]
```diff
@@ -117,14 +117,17 @@ public function toString($indent = null)
             ? $this->getWhitespace($indent)
             : $this->getIndent();
 
-        $items  = $this->getArrayCopy();
         // If we don't have items - do not show prefix and postfix
-        if (!count($items)) {
+        if (!$this->count()) {
```
Comment (froschdesign) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: Please move this check before `$indent = ($indent !== null)` (line 116) and add a whitespace after `!`.

File: src/Helper/Placeholder/Container/AbstractContainer.php [hunk 3]
```diff
@@ -117,14 +117,17 @@ public function toString($indent = null)
...
             return '';
         }
+
+        $itemsToString = implode($this->getSeparator(), $this->getArrayCopy());
+        // todo check empty $itemsToString after trim() to return ''
```
Comment (froschdesign) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: If items with whitespace are added, then there is a reason for it and we should allow the rendering. Please undo this changes.

File: src/Helper/Placeholder/Container/AbstractContainer.php [hunk 4]
```diff
@@ -113,21 +113,19 @@ public function __toString()
...
-            : $this->getIndent();
-
         // If we don't have items - do not show prefix and postfix
         if (!$this->count()) {
             return '';
         }
 
-        $itemsToString = implode($this->getSeparator(), $this->getArrayCopy());
-        // todo check empty $itemsToString after trim() to return ''
+        $indent = ($indent !== null)
```
Comment (malukenho) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: May be we should avoid the negative check:

```php
$indent = $indent === null
    ? $this->getIndent()
    : $this->getWhitespace($indent);
```

File: src/Helper/Placeholder/Container/AbstractContainer.php [hunk 5]
```diff
@@ -113,8 +113,7 @@ public function __toString()
      */
     public function toString($indent = null)
     {
-        // If we don't have items - do not show prefix and postfix
```
Comment (froschdesign) [inline src/Helper/Placeholder/Container/AbstractContainer.php]: The comment was good.

**Comment by weierophinney** [comment]
Thanks, @AndyDune; merged to develop for release with 2.10.0, as it is a slight change in behavior.
